implementation.js

```diff
@@ -1,6 +1,93 @@
 'use strict';
 var ES = require('es-abstract/es6');
 var supportsDescriptors = require('define-properties').supportsDescriptors;
+var has = require('has');
+var global = require('system.global')();
+
+var parseIterable = function (iterator) {
+	var done = false;
+	var iterableResponse;
+	var tempArray = [];
+
+	if (iterator && typeof iterator.next === 'function') {
+		while (!done) {
+			iterableResponse = iterator.next();
+			if (
+				has(iterableResponse, 'value')
+				&& has(iterableResponse, 'done')
+			) {
+				if (iterableResponse.done === true) {
+					done = true;
+					break; // eslint-disable-line no-restricted-syntax
+
+				} else if (iterableResponse.done !== false) {
+					break; // eslint-disable-line no-restricted-syntax
+				}
+
+				tempArray.push(iterableResponse.value);
+			} else if (iterableResponse.done === true) {
+				done = true;
+				break; // eslint-disable-line no-restricted-syntax
+			} else {
+				break; // eslint-disable-line no-restricted-syntax
+			}
+		}
+	}
+
+	return done ? tempArray : false;
+};
+
+var hasSymbols = typeof Symbol === 'function' && typeof Symbol.iterator === 'symbol';
+var iteratorSymbol;
+if (hasSymbols) {
+	iteratorSymbol = Symbol.iterator;
+} else {
```

Comment:
you are still missing a third case - some versions of Safari, for instance, don't have `Symbol.iterator`, but do have Maps/Sets that are iterable only via `for..of`. In this case, we'd need to iterate over the collection via `for..of` in a dynamically created `Function`.

See https://github.com/ljharb/map-tojson/blob/master/index.js#L11-L29 for some hoops I've jumped through in the past (it might be useful to make a separate "iterate" package to encapsulate all of this logic)